### PR TITLE
Tests: Shift all tests to use no globals

### PR DESCRIPTION
**Pull request by apsdehal**
This PR shifts all the tests to use no global QUnit variables and methods.

### CODE REVIEW

**Comment by jzaefferer** [comment]
For the Intern migration this looks good. For the full migration to the QUnit 2.0 API its still missing asyncTest(), which I've filed an issue for in the underlying tool: https://github.com/apsdehal/qunit-migrate/issues/4

We still have a work-in-progress PR that changes all test defines (#1569) and one that changes a ton of tests for one widget (#1513). @arschmitz tells me that the changes here are all automated, so to avoid merge conflicts we should land the other two PRs first (should happen in the next two weeks), then apply the automatic changes once more to do the migration.

**Comment by jzaefferer** [comment]
@apsdehal we've landed the button rewrite in master. And you've addressed the `asyncTest()` migration. Could you update this PR based on the current master branch? Thanks.

**Comment by apsdehal** [comment]
Will do it ASAP.

**Comment by scottgonzalez** [comment]
@apsdehal I see you pushed some updated a few days ago. Is this now ready for another review?

**Comment by apsdehal** [comment]
@scottgonzalez No, not yet. I will ping you whenever this is ready.

**Comment by apsdehal** [comment]
@scottgonzalez It is ready for a round of review. Only test failing is one in effects, which if I change to asyncTest back works smoothly.

**Comment by jzaefferer** [comment]
The issue is in the ".hide() with step" test in effects/core.js, which passes `start` as a complete callback. Needs to be replaced.

**Comment by apsdehal** [comment]
@jzaefferer Issue has been fixed now and the build is passing.

File: tests/unit/effects/core.js [hunk 1]
```diff
@@ -132,18 +137,19 @@ asyncTest( "animateClass works with colors", function() {
...
-			present( test.css( "backgroundColor" ), [ "#ffffff", "#fff", "rgb(255, 255, 255)" ], "Color is final" );
-			start();
+			assert.present( test.css( "backgroundColor" ), [ "#ffffff", "#fff", "rgb(255, 255, 255)" ], "Color is final" );
+			ready();
 		}
 	} );
 } );
 
-asyncTest( "animateClass calls step option", 1, function() {
+QUnit.test( "animateClass calls step option", 1, function( assert ) {
```
Comment (jzaefferer) [inline tests/unit/effects/core.js]: Could you convert the expect argument to `assert.expect` calls as well?

Comment (jzaefferer) [inline tests/unit/effects/core.js]: Could you convert the expect argument to assert.expect calls as well?

Comment (jzaefferer) [inline tests/unit/effects/core.js]: Could you convert the expect argument to assert.expect calls as well? (3rd attempt)

**Comment by jzaefferer** [comment]
I don't now why GitHub instantly marks by diff comment as outdated. So:

> +QUnit.test( "animateClass calls step option", 1, function( assert ) {

Could you convert the expect argument to assert.expect calls as well?

**Comment by apsdehal** [comment]
@jzaefferer Fixed.

**Comment by jzaefferer** [comment]
I guess those 2 instances were the only ones still in master, nice.

**Comment by apsdehal** [comment]
Yes, I ran a regex check on the whole repo. Those were the only two.

File: tests/lib/common.js [hunk 2]
```diff
@@ -87,9 +88,10 @@ exports.testJshint = function( module ) {
 		return;
 	}
 
-	asyncTest( "JSHint", function() {
-		require( [ "jshint" ], function() {
-			expect( 1 );
+	QUnit.test( "JSHint", function( assert ) {
+        var ready = assert.async();
+        require( [ "jshint" ], function() {
```
Comment (scottgonzalez) [inline tests/lib/common.js]: Convert indentation to tabs.

**Comment by scottgonzalez** [comment]
Please add a commit at the end that removes all the QUnit globals from https://github.com/jquery/jquery-ui/blob/435ed7e8d2e0221635494eedfbe22616f2a4814f/tests/.jshintrc#L20

**Comment by apsdehal** [comment]
@scottgonzalez I have addressed your comments.

**Comment by jzaefferer** [comment]
Testing against custom QUnit build (pre-2.0) based on https://github.com/jquery/qunit/pull/976 

`tests/unit/all.html` isn't working due to https://github.com/JamesMGreene/qunit-composite/issues/9

At least `tests/unit/dialog/dialog.html` and `tests/unit/effects/effects.html` fail due to https://github.com/JamesMGreene/qunit-assert-close/issues/4

There might be other failures, but its hard to tell with the many errors coming from `qunit-assert-close`.

@apsdehal would you be interested in patching qunit-composite, qunit-assert-close or both? That would help a lot.

**Comment by apsdehal** [comment]
@jzaefferer Sure, will do it!

**Comment by jzaefferer** [comment]
@scottgonzalez updating those externals shouldn't block this, after all the PR works just fine, and QUnit 2.0 isn't even released. Based on my testing with QUnit pre-2.0, the changes here work fine. Unless I'm missing something, we can merge this and release rc2. (I'm not sure why I thought that rc2 should wait).

**Comment by scottgonzalez** [comment]
@jzaefferer I'm not sure what externals you're referring to. Did you mean the globals? @apsdehal already removed them from the JSHint config. I asked for that to be part of this PR to ensure that none of the globals were missed during the update.

**Comment by jzaefferer** [comment]
Thanks a lot @apsdehal!

**Pull request merged**